src/UsgsAstroLsSensorModel.cpp

```diff
@@ -1938,6 +1941,25 @@ void UsgsAstroLsSensorModel::getAdjSensorPosVel(const double& time,
   double sensPosNom[3];
   lagrangeInterp(m_numPositions / 3, &m_positions[0], m_t0Ephem, m_dtEphem,
                  time, 3, nOrder, sensPosNom);
+
+  // Avoid computing the velocity and adjustments, if not needed, as those
+  // take up at least half of this function's time.
+  bool has_adj = false;
+  for (size_t it = 0; it < adj.size(); it++) {
+    if (adj[it] != 0) {
+      has_adj = true;
+    }
+  }
+  if (!has_adj && !calc_vel) {
+    xc = sensPosNom[0];
+    yc = sensPosNom[1];
+    zc = sensPosNom[2];
+    vx = 0.0;
+    vy = 0.0;
+    vz = 0.0;
+    return;
```

Comment:
Add a logging message here before the return

Reply:
Added.